src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,468 @@
+  private static UUID instanceId;
+  private static UUID mainLibraryLocationId;
+  private static UUID annexLibraryLocationId;
+  private static UUID onlineLocationId;
+  private static UUID secondFloorLocationId;
+
+  @BeforeClass
+  public static void beforeAny() throws Exception {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+
+    StorageTestSuite.deleteAll(materialTypesStorageUrl(""));
+    StorageTestSuite.deleteAll(locationsStorageUrl(""));
+    StorageTestSuite.deleteAll(locLibraryStorageUrl(""));
+    StorageTestSuite.deleteAll(locCampusStorageUrl(""));
+    StorageTestSuite.deleteAll(locInstitutionStorageUrl(""));
+    StorageTestSuite.deleteAll(loanTypesStorageUrl(""));
+
+    MaterialTypesClient materialTypesClient = new MaterialTypesClient(client, materialTypesStorageUrl(""));
+    journalMaterialTypeID = materialTypesClient.create("journal");
+    canCirculateLoanTypeID = new LoanTypesClient(client, loanTypesStorageUrl("")).create("Can Circulate");
+
+    instanceId = instancesClient.create(instance(UUID.randomUUID())).getId();
+
+    LocationsTest.createLocUnits(true);
+    mainLibraryLocationId = LocationsTest.createLocation(null, MAIN_LIBRARY_LOCATION, "It/M");
+    annexLibraryLocationId = LocationsTest.createLocation(null, ANNEX_LIBRARY_LOCATION, "It/A");
+    onlineLocationId = LocationsTest.createLocation(null, ONLINE_LOCATION, "It/O");
+    secondFloorLocationId = LocationsTest.createLocation(null, SECOND_FLOOR_LOCATION, "It/SF");
+    LocationsTest.createLocation(null, THIRD_FLOOR_LOCATION, "It/TF");
+    LocationsTest.createLocation(null, FOURTH_FLOOR_LOCATION, "It/FoF");
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs("item");
+    StorageTestSuite.checkForMismatchedIDs("holdings_record");
+  }
+
+  public void canCalculateEffectiveLocationOnHoldingUpdate() throws Exception {
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    final Item[] itemsToCreate = {
+      buildItem(holdingsRecordId, null, null),
+      buildItem(holdingsRecordId, null, null),
+      buildItem(holdingsRecordId, null, null)
+    };
+
+    for (Item item : itemsToCreate) {
+      IndividualResource createdItem = createItem(item);
+      assertTrue(createdItem.getJson().containsKey("effectiveLocationId"));
+    }
+
+    JsonObject holding = holdingsClient.getById(holdingsRecordId).getJson();
+    holding.put("temporaryLocationId", secondFloorLocationId.toString());
+    holdingsClient.replace(holdingsRecordId, holding);
+
+    for (Item item : itemsToCreate) {
+      Item fetchedItem = getItem(item.getId());
+      assertEquals(fetchedItem.getEffectiveLocationId(), secondFloorLocationId.toString());
+    }
+  }
+
+  @Test
+  public void canCalculateEffectiveLocationOnHoldingRemoveTempLocation() throws Exception {
```

Comment:
Can these test names describe what effective location is being calculated, e.g. that it should be the permanent location from the holding.

Reply:
done